docs/cloudevents/cloudevents-spans.md

```diff
@@ -35,168 +31,38 @@ consult the
-|    |     +---------------------------------------+    |
-|    |     | Send (auto-instr)                     |    |
-|    |     +---------------------------------------+    |
-|    |        +------------------------------------+    |
-|    |        | Intermediary: Received (auto-instr)|    |
-|    |        +------------------------------------+    |
-|    |        +------------------------------------+    |
-|    |        | Intermediary: Send (auto-instr)    |    |
-|    |Link    +------------------------------------+    |
-|    |                                                  |
-|    |                                                  |
-|    |                                                  |
-|    |     Trace 2                                      |
-|    |                                                  |
-|    |     +---------------------------------------+    |
-|    |     | Consumer: Receive (auto-instr)        |    |
-|    |     +---------------------------------------+    |
-|    |       +-------------------------------------+    |
-|    +------ | Consumer: Process                   |    |
-|            +-------------------------------------+    |
-|                                                       |
-+-------------------------------------------------------+
-```
-
 ## Conventions
 
-To achieve the trace above, it is necessary to capture the context of
-the event creation so that when the CloudEvent reaches its destination(s), this
-context can be continued. Each CloudEvent acts then as the medium of this
-context propagation.
-
-This document relies on the CloudEvents specification, which defines this
-context propagation mechanism via the
-[CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md).
-Once the trace context is set on the event
-via the Distributed Tracing Extension, it MUST not be modified.
-
-The remainder of this section describes the semantic conventions for Spans
-required to achieve the proposed trace.
-
-### Spans
-
-#### Creation
-
-Instrumentation SHOULD create a new span and populate the
-[CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md)
-on the event. This applies when:
-
-- A CloudEvent is created by the instrumented library.
-  It may be impossible or impractical to create the Span during event
-  creation (e.g. inside the constructor or in a factory method),
-  so instrumentation MAY create the Span later, when passing the event to the transport layer.
-- A CloudEvent is created outside of the instrumented library
-  (e.g. directly constructed by the application owner, without calling a constructor or factory method),
-  and passed without the Distributed Tracing Extension populated.
+CloudEvents can be sent via various transports, such as HTTP or via messaging
+systems. Each transport defines its own mechanism of
```

Comment:
I left a comment on https://github.com/cloudevents/spec/pull/912#issuecomment-2195429167 - not each protocol. 

There are at least 2 Azure messaging services that don't have any place to stick trace context to. 

I support removing context propagation section above, but here I'd say  something like

When CloudEvents are used in messaging scenarios and corresponding messaging system supports propagating trace context along with the message independently of CloudEvents, the corresponding messaging instrumentation SHOULD propagate the context using this messaging-system capabilities.
Such instrumentations may not be aware of the payload format or could support CloudEvents along with other formats.

If CloudEvents are instrumented independently of messaging, the CloudEvent-specific instrumentation SHOULD use CloudEvents distributed context propagation extension to inject the context and is encouraged to follow span structure described in messaging semantic conventions.

Reply:
Right, I think this is very close to what I tried to say below:

> When the transport being used does not have a well established
propagation mechanism, instrumentations MAY rely on the
[CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md).
as means to propagate the trace context, while still following either the HTTP
or Messaging span semantic conventions.

I think we are aligned on the objective. I will try to rephrase it a bit and incorporate some of your suggestions :)

Reply:
It seems also you want to leave out completely the mention about CloudEvents used in HTTP scenarios?

Reply:
I made some changes, let me know what you think!